src/app/asses-reports/asses-reports.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, OnInit } from '@angular/core';
+
+import { authAppState } from '../interfaces/responses.interfaces';
+import { Router } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { selectUserAssessments } from '../store/auth.selectors';
+import * as AuthActions from '../store/auth.actions';
+
+@Component({
+  selector: 'app-asses-reports',
+  templateUrl: './asses-reports.component.html',
+  styleUrls: ['./asses-reports.component.css'],
+})
+export class AssesReportsComponent implements OnInit {
+  assessments$ = this.store.select(selectUserAssessments);
+
+  constructor(private router: Router, private store: Store<authAppState>) {}
+
+  ngOnInit(): void {
+    if (localStorage.getItem('token') !== null) {
+      this.store.dispatch(AuthActions.getAssessments());
+    }
```

Comment:
you already have guard, so it's impossible to go to this component if token is null